src/mlpack/methods/neighbor_search/neighbor_search.hpp

```diff
@@ -413,18 +500,21 @@ class NeighborSearch
   size_t Scores() const { return scores; }
 
   //! Access whether or not search is done in naive linear scan mode.
+  //! Deprecated. Will be removed in mlpack 3.0.0.
```

Comment:
Hm, I dunno, maybe we should update this comment.  Someone might see it and say "but what are we supposed to use once these methods are gone?", but we don't have any advice for them there.  I know that we can't add `SearchMode()` until 3.0.0 though.  So I guess possibly we could just point out that `SearchMode()` will be available in 3.0.0 and will be the replacement for these functions.